### ISSUE TITLE
Header not accessible with screen reader on iOS with headerMode='screen' and headerTransparent: true

### ISSUE THREAD

**Issue by jangsoori** [react-navigation/react-navigation#9905]
### Current behavior

When you try to move around with accessibility gesture throughout the screen with voice over turned on (swipe left, right), there is no way to get to the header for the user that uses screen reader. It happens when screen has `headerTransparent: true` option and navigator is set to `headerMode='screen'`.

Tap on the text on the snack, and try to swipe left to move to the header. When you set `headerMode='float'` it works, but with `screen` it does not.

### Expected behavior

While using accessibility gestures of voice over (swipe left, right), user should be able to access a header and its content.

### Reproduction

https://snack.expo.dev/@jangsoori/stack

### Platform

- [ ] Android
- [X] iOS
- [ ] Web
- [ ] Windows
- [ ] MacOS

### Packages

- [ ] @react-navigation/bottom-tabs
- [ ] @react-navigation/drawer
- [ ] @react-navigation/material-bottom-tabs
- [ ] @react-navigation/material-top-tabs
- [X] @react-navigation/stack
- [X] @react-navigation/native-stack

### Environment

| package                                | version |
| -------------------------------------- | ------- |
| @react-navigation/native               |  6.0.2
| @react-navigation/stack                | 6.0.7
| react-native-safe-area-context         | 3.2.0
| react-native-screens                   | 3.4.0
| react-native-gesture-handler           | 1.10.2
| react-native-reanimated                | 2.2.0
| react-native                           |  0.63.3

**Comment by tr3v3r** [comment]
Also with the same settings, headerLeft and headerRight items are not clickable on Android! 
If I set headerMode='float' the buttons become responsive

**Comment by LuigiClaudio** [comment]
I can see the same issue on Android using TalkBack. With `headerTransparent: true` and TalkBack on (no matter the `headerMode` being this `screen` or `float`), the elements in the header are not accessible neither by swiping nor by tapping.

**Comment by Shanghai-** [comment]
Confirming that this issue is still present on both iOS and Android with these settings. Using the "float" headerMode works as a workaround for the issue, but a true fix would be a huge improvement for the vision-impaired community, as many developers are likely using the default "screen" mode on Android without realizing that it breaks their screen reader access entirely.

Digging in to the TalkBack logs a bit on Android, I found that any buttons I put in the "screen" header are given improper bounds:
`(704697)315.Button:invisible:(42, 148 - 147, 0):CONTENT{Back (action:FOCUS/A11Y_FOCUS/CLICK):focusable:clickable`

The back button here starts at (42, 148) and ends at (147, 0), which causes TalkBack to register it as "invisible" due to its invalid bounds (start should be < end in both dimensions) and hide it from users. In "screen" mode, the same button is given the correct bounds, and functions properly with TalkBack. I haven't dug into the react-navigation codebase yet to figure out why this is.

**Comment by t1gu1** [comment]
With `@react-navigation/native-stack` screenreader on IOS has the same problem.

**Comment by AdamBowler** [comment]
Getting the same issue with all our headers, also tab bars are completley inaccessible and headerMode float doesn't seem to help

**Comment by hraschan** [comment]
Is there any progress on this topic?

**Comment by terreng** [comment]
I was having this issue when using a custom header component with native-stack, and in my case the issue was the use of `position: absolute` on the component. Removing that and using `headerTransparent: true` instead achieved the same result and fixed the problem.
